docs/core-concepts/atomic-swap-htlc.md

```diff
@@ -0,0 +1,133 @@
+
+## HTLC
+
+HTLC stands for Hash Time Locked Contract. An HTLC is a script or smart contract that locks a digital asset over time.
+HTLCs require a ledger that provides a time lock.
+
+<iframe 
+    src="https://docs.google.com/presentation/d/e/2PACX-1vRYb97VhvfyMa3oFC8CiVagBrNmOioSJpasERtSCi6RMwf0MwxCb1yTJBUm8_ZT9OOw6r-_fBsV23GX/embed?start=false&loop=false&delayms=3000"
+    frameborder="0"
+    width="801"
+    height="480"
+    allowfullscreen="true"
+    mozallowfullscreen="true"
+    webkitallowfullscreen="true">
+</iframe>
+
+HTLCs typically have three steps:
+
+1. **Fund**: Locking up the asset in the HTLC.
+1. **Redeem**: Until the expiry time-stamp is reached the other party has the change to redeem the locke-up assets with the secret.
+1. **Refund**: After the expiry condition has been met, the party that locked up the asset can take it back.
+
+In order to create HTLCs for the execution of an atomic swap, the two parties have to exchange certain parameters:
+
+1. Ledgers, assets and amounts to be swapped.
+1. Redeem and refund identity (address) of the respective parties.
+1. Expiry times to be used for the HTLCs.
+1. Hash of the secret (to be sent from the party in the cryptographic role of Alice to the party in the cryptographic role of Bob).
+
+### HTLCs on different Ledgers
+
+On Bitcoin this can be achieved by using a [Bitcoin script](https://en.bitcoin.it/wiki/Script).
+The Bitcoin HTLC Script is defined in the [COMIT Bitcoin HTLC specification](https://github.com/comit-network/RFCs/blob/master/RFC-005-SWAP-Basic-Bitcoin.adoc): 
+
+```
+OP_IF
+    OP_SIZE 32 OP_EQUALVERIFY
+    OP_SHA256 <secret_hash> OP_EQUALVERIFY
+    OP_DUP OP_HASH160 <redeem_identity>
+OP_ELSE
+    <refund_timestamp> OP_CHECKLOCKTIMEVERIFY OP_DROP
+    OP_DUP OP_HASH160 <refund_identity>
+OP_ENDIF
+OP_EQUALVERIFY
+OP_CHECKSIG
+```
+
+On Ethereum this can be achieved by deploying an [Ethereum Smart Contract](https://solidity.readthedocs.io/en/v0.6.2/introduction-to-smart-contracts.html).
+Note that an Ethereum HTLC does not necessarily have to be written in Solidity.
+Given that HTLCs are really simple smart contracts, they can also be written in EVM assembly as defined in the [COMIT Ether HTLC specification](https://github.com/comit-network/RFCs/blob/master/RFC-007-SWAP-Basic-Ether.adoc) and [COMIT ERC20 HTLC specification](https://github.com/comit-network/RFCs/blob/master/RFC-009-SWAP-Basic-ERC20.adoc).
+
+[Scriptless Scripts](privacy-preserving-swap.md#scriptless-scripts) enable HTLCs hidden in signatures. This allows [Privacy Preserving Swaps](privacy-preserving-swap.md).
+
+### Close Look on HTLC Expiries
+
+The expiries of the HTLCs have to be defined so that Alice's expiry time is further in the future than Bob's.
+This is because Alice is protected by the secret, and if she choses not to reveal the secret, Bob must have the change to refund early.
```

Comment:
```suggestion
This is because Alice is protected by the secret, and if she chooses not to reveal the secret, Bob must have the chance to refund early.
```